hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HAUtils.java

```diff
@@ -389,18 +391,25 @@ public static List<String> buildCAList(CertificateClient certClient,
       if (!SCMHAUtils.isSCMHAEnabled(configuration)) {
         return generateCAList(certClient);
       } else {
+        Set<X509Certificate> rootCAs = certClient.getAllRootCaCerts();
+        if (!rootCAs.isEmpty()) {
+          List<String> rootCaCertPems = new ArrayList<>();
+          for (X509Certificate rootCA : rootCAs) {
+            rootCaCertPems.add(CertificateCodec.getPEMEncodedString(rootCA));
+          }
+          return rootCaCertPems;
+        }
         Collection<String> scmNodes = SCMHAUtils.getSCMNodeIds(configuration);
-        int expectedCount = scmNodes.size() + 1;
```

Comment:
Thinking this through a bit, this change seems to be an interesting one.

This check was useful, as the old trust model was based on getting all the sub-CA certificate and the rootCA certificate, and then validating single certificates based on that. In our new system, we rely only on the rootCA certificate.

But when a current client talks to an older cluster, we still need to get all the certificates. And in this case the certClient.getAllRootCaCerts() method is unknown on the server side, and I guess we should get an exception to handle, and then we should fallback to the old behaviour with the old check...

We discussed about this change, but since then I realized that it is not enough to just simplify the code here, but we need to keep the possibility to talk to an older cluster where we have the old system.